Nef_3/include/CGAL/Nef_3/SNC_decorator.h

```diff
@@ -342,172 +342,20 @@ class SNC_decorator : public SNC_const_decorator<Map> {
     return( e1!=e2 && SD.cyclic_adj_succ(e2)==e1);
   }
 
-  Halffacet_handle get_visible_facet( const Vertex_handle v,
-                                      const Ray_3& ray) const
-    /*{\Mop when one shoot a ray |ray| in order to find the facet below to
-      an object, and vertex |v| is hit, we need to choose one of the facets
-      in the adjacency list of |v| such that it could be 'seen' from the
-      piercing point of the |ray| on the sphere map on |v|.  We make it just
-      locating the sphere facet |sf| pierced by |ray| and taking the adjacent
-      facet to one of the sphere segments on the boundary of |sf|.
-      \precondition |ray| target is on |v| and the intersection between
-      |ray| and the 2-skeleton incident to v is empty. }*/ {
+  template <typename Traits = Self::Decorator_traits>
```

Comment:
For this and the next functions it seems also that you can fully remove the template and hardcode the return type, and then call the function from `Base`  with `Decorator_traits`  without `Self::`

Reply:
I think I've already made this change in: https://github.com/CGAL/cgal/pull/5699/commits/77a03f2084e9f6f78527f45144bc3d95ab2fb4c8